src/converse-chatview.js

```diff
@@ -344,11 +344,19 @@
                                "Output has been shortened."),
                             true, true);
                     }
+
+                    var mess_time;
+
+                    if(_converse.time_format === 'h')
+                        mess_time = msg_time.format('hh:mm a');
+                    else if(_converse.time_format === 'H')
+                        mess_time = msg_time.format('HH:MM');
+
                     var $msg = $(template(
                         _.extend(this.getExtraMessageTemplateAttributes(attrs), {
                             'msgid': attrs.msgid,
                             'sender': attrs.sender,
-                            'time': msg_time.format('hh:mm'),
+                            'time': mess_time,
```

Comment:
@jcbrand in one of the above comments, you have told to keep the original one(which according to current release is hh:mm) as the default. Can you confirm once again, which one will I finally keep .

Moreover, doing the above check helps in printing am or pm beside the time in 12hr format. Because for that moment needs hh:mm a. One way is of-course to let the user specify 'hh:mm a' as the setting. And the other way would be to simply leave the printing of 'am' or 'pm' in messages, which I feel makes a message sent at 6am and 6pm confusing.